scipy/optimize/linprog_ip.py

```diff
@@ -0,0 +1,1994 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Thu Feb 02 17:09:17 2017
+
+@author: Matt
+
+To do:
+multiple presolve
+questions for scipy-dev
+* This is written in Python 2. How should I check compatibility with Python 3?
+* Are the warnings OK? Do I need an option to disable them?
+* Is 100% pep8 compliance necessary?
+* How should unit tests be performed on both simplex and ip?
+* Can I leave multi-pass presolve to an update?
+* Can I leave sparse presolve to an update?
+* Do I need to implement a callback interface?
+* There are some changes to be made to simplex; it should use the ip presolve
+ routine. Can that be left to an update?
+* How do I document options not common with simplex?
```

Comment:
You should write your entry point routine as
```
def _linprog_ip(c, A_ub=None, b_ub=None, A_eq=None, b_eq=None,
            bounds=None, method='simplex', callback=None,
            ip_option_1=456, ip_option_2=123, **unknown_options):
    _check_unknown_options(unknown_options)
    ...
```
where `ip_option_1=456, ip_option_2=123` should be replaced with the ip-specific options you support. Documentation of these options goes to the docstring of `_linprog_ip` under a separate `Options` header instead of under `Parameters`.

You should create a file `doc/source/optimize.linprog-ip.rst` with similar content as `optimize.linprog-simplex.rst` and edit the docstring in `scipy/optimize/__init__.py` accordingly.

In the `def linprog` routine in `_linprog.py` you should add (exactly) the following lines:
```
    elif meth == 'ip':
        return _linprog_ip(c, A_ub=A_ub, b_ub=b_ub, A_eq=A_eq, b_eq=b_eq,
                                bounds=bounds, callback=callback, **options)
```
The tests testing the solver should then run via `linprog(..., method="ip")`.

I'm not sure if "ip" should be the name of the solver --- a more specific name is probably better.